packages/js/src/classic-editor/replacement-variables/configurations.js

```diff
@@ -45,7 +45,22 @@ export const parentTitle = {
 export const primaryCategory = {
 	name: "primary_category",
 	getLabel: () => __( "Primary category", "wordpress-seo" ),
-	getReplacement: () => get( window, "wpseoScriptData.analysis.plugins.replaceVars.replace_vars.primary_category", "" ),
+	getReplacement: () => {
+		// Gets the primary category data from `yoast-seo/editor` store.
+		const primaryCategoryData = select( "yoast-seo/editor" ).getReplaceVars()
+			.filter( replaceVar => replaceVar.name === "primary_category" );
+
+		/*
+		 * The `yoast-seo/editor` store returns an empty primary category upon opening a content
+		 * even when there is already a primary category set to it.
+		 * When this happens, we retrieve the primary content from wpseoPrimaryCategoryL10n object.
+		 */
+		const initialPrimaryCat = get( window, "wpseoPrimaryCategoryL10n.taxonomies.category" );
+		const termId = initialPrimaryCat.primary;
+		const termName = initialPrimaryCat.terms.filter( term => term.id === termId )[ 0 ]?.name;
```

Comment:
For the reviewers: the primary category retrieved from `wpseoPrimaryCategoryL10n` object is used as a fallback value. Somehow, when opening a content, the `yoast-seo/editor` store returns an empty primary category, even when there is already a primary category set to the content. Also feel free to leave a comment if you know a better solution 😺

Reply:
I would select the categories from the SEO store:
```js
const categories = select( SEO_STORE_NAME ).selectCategories();
```
And then find the primary category, based on the primary category ID:
```js
const primaryCategoryName = categories.find( cat => cat.id === primaryTaxonomyId.toString() ).name;
```

Reply:
Thanks @hansjovis ! I've implemented your suggestion and everything still works as expected! 🙌